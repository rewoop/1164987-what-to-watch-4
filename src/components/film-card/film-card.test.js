import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card.jsx";

const filmInfo = {
  FILM_TITLE: `Fantastic Beasts`,
  FILM_IMAGE: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  FILM_VIDEO: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

it(`Should FilmCard render correctly`, () => {
  const tree = renderer
    .create(<FilmCard
      film={filmInfo}
      onTitleClickHandler={() => {}}
      onPosterClickHandler={() => {}}
      isPlaying={false}
      setPlayingFilm={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
