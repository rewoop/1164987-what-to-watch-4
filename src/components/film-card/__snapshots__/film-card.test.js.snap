// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/film"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      />
      <img
        alt="Fantastic Beasts"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film"
      onClick={[Function]}
    >
      Fantastic Beasts
    </a>
  </h3>
</article>
`;
